### PR TITLE
Allow adding names with comma or hyphen #110

### LINKED ISSUE #110: Add doesn't allow adding names with comma or hyphen
The regex is too restrictive (`[\\p{Alnum} ]+`)

### PR DESCRIPTION
**Pull request by okkhoy**
Fixes #110

### CODE REVIEW

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 1]
```diff
@@ -392,6 +407,17 @@ Person adam() throws Exception {
...
+        Person borg() throws Exception {
+            Name name = new Name("Bj�rn o'Borg Jos� John-Doe Jr.");
+            Phone privatePhone = new Phone("98761234");
+            Email email = new Email("borg@jose.com");
+            Address privateAddress = new Address("12, heere at the wall");
+            Tag tag1 = new Tag("unicodeName");
+            Tag tag2 = new Tag("HyphenDotQuote");
+            UniqueTagList tags = new UniqueTagList(tag1, tag2);
+            return new Person(name, privatePhone, email, privateAddress, tags);
+        }
```
Comment (damithc) [inline src/test/java/seedu/address/logic/LogicManagerTest.java]: Perhaps change an existing test user in the TestDataHelper? That will ensure we automatically test unicode in other places as well.

**Comment by damithc** [comment]
@ndt93 can you review this one?

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 2]
```diff
@@ -189,6 +189,21 @@ public void execute_add_successful() throws Exception {
     }
 
     @Test
+    public void execute_add_differentCharactersInName_successful() throws Exception{
```
Comment (damithc) [inline src/test/java/seedu/address/logic/LogicManagerTest.java]: `executeAdd` (the name convention has only three parts)

**Comment by ndt93** [comment]
@damithc Sure.

File: src/main/java/seedu/address/model/person/Name.java [hunk 3]
```diff
@@ -9,7 +9,12 @@
 public class Name {
 
     public static final String MESSAGE_NAME_CONSTRAINTS = "Person names should be spaces or alphanumeric characters";
-    public static final String NAME_VALIDATION_REGEX = "[\\p{Alnum} ]+";
+    public static final String NAME_VALIDATION_REGEX = "^[\\p{L}0-9 .'-]+$"; 
+    // In the above:
+    // \\p{L} matches any unicode letter (like German name with accents)
+    // . used to match dots in the name E.g., John Paul Jr.
+    // ' is used to match names like d'Souza
+    // - is used to match names like Jolie-Pitt
```
Comment (ndt93) [inline src/main/java/seedu/address/model/person/Name.java]: Do we have any convention for class fields' comments in 2103 (above or below the field declaration etc.)? @damithc

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 4]
```diff
@@ -189,6 +189,21 @@ public void execute_add_successful() throws Exception {
...
+        // setup expectations
+        TestDataHelper helper = new TestDataHelper();
+        Person toBeAdded = helper.borg();
+        AddressBook expectedAB  = new AddressBook();
+        expectedAB.addPerson(toBeAdded);
+        
+        // execute command and verify the result
```
Comment (ndt93) [inline src/test/java/seedu/address/logic/LogicManagerTest.java]: These in-method comments shouldn't be necessary?

File: src/test/java/seedu/address/logic/LogicManagerTest.java [hunk 5]
```diff
@@ -392,6 +407,19 @@ Person adam() throws Exception {
             UniqueTagList tags = new UniqueTagList(tag1, tag2);
             return new Person(name, privatePhone, email, privateAddress, tags);
         }
+        
+        Person borg() throws Exception {
+            Name name = new Name("Bj\u00F6rn o'Borg Jos\u00E9 John-Doe Jr.");
+            // names given in unicode format based on the thread: 
+            // http://stackoverflow.com/questions/4237581/comparing-unicode-characters-in-junit
```
Comment (ndt93) [inline src/test/java/seedu/address/logic/LogicManagerTest.java]: Maybe better to put these comments in the `borg` method's header description instead.

File: src/test/java/seedu/address/logic/LogicManagerTest.java (hunk 4 above)
Comment (damithc) [inline src/test/java/seedu/address/logic/LogicManagerTest.java]: Ok to have in this case (they seem to tell the high-level story).

**Comment by damithc** [comment]
Closing due to inactivity.

**Pull request closed without merging**